### PR TITLE
Altered README format for modules

### PR DESCRIPTION
**Pull request by JoeCSykes**

### CODE REVIEW

**Comment by robg-answer** [comment]
FYI this'll be conflicting with the Terraform fmt change as part of CI.
I'd suggest not bother trying to merge take yours and run terraform fmt on it.

File: Terraform_modules/vpc_subnets/README.md [hunk 1]
```diff
@@ -7,93 +7,85 @@ See image for an example structure when the region in the provider is set to `eu
...
+the second uses the module to create 1 public subnet in the AZ `eu-west-1` and 2 private subnets in `eu-west-1` 
+and `eu-west-3` respectively. Note that when calling the module directly from the github
+repository you can specify a version by appending the below source reference with 
+`?ref=v1.2.0` for version "1.2.0" (for further information please see 
+[here](https://developer.hashicorp.com/terraform/language/modules/sources#modules-in-package-sub-directories))
+
+<pre><code>
+
+module "vpc_subnet" {
+  source       = "git::https://github.com/AnswerConsulting/AnswerKing-Infrastructure.git//Terraform_modules/rds_serverless_cluster?ref=v1.0.0"
```
[the remark this answers is not in the document]
Comment (JoeCSykes) [reply Terraform_modules/vpc_subnets/README.md]: changed to reference vpc_subnet module

File: Terraform_modules/vpc_subnets/variables.tf [hunk 2]
```diff
@@ -1,54 +1,77 @@
...
 }
 
 variable "num_private_subnets" {
   type        = number
-  description = "Number indicating how many public subnets you want"
+  description = <<EOT
+                This is a number specifying how many private subnets you want. Not specifying this will result
+                in `x` private subnets where `x` is the number of az zones. If the number specified is greater than
+                the number of Availability Zones (AZs) the public subnets will be spread out evenly over the
+                available AZs. The CIDR values used are of the form `10.0.{i}.0/24` where `i` starts at 1 and
```
Comment (cmbuckley) [inline Terraform_modules/vpc_subnets/variables.tf:61]: `i` starts at 101

**Review by cmbuckley — CHANGES_REQUESTED** [review]
Not part of the diff here, but I would advise marking the username/password outputs of rds_serverless_cluster as `sensitive = true`.

File: Terraform_modules/vpc_subnets/variables.tf [hunk 3]
```diff
@@ -1,54 +1,77 @@
...
 
 # Subnet variables
 variable "num_public_subnets" {
   type        = number
-  description = "Number indicating how many public subnets you want"
+  description = <<EOT
+                This is a number specifying how many public subnets you want. Not specifying this will result
+                in `x` public subnets where `x` is the number of az zones. If the number specified is greater than
+                the number of Availability Zones (AZs) the public subnets will be spread out evenly over the
+                available AZs. The CIDR values used are of the form `10.0.{i}.0/24` where `i` starts at 1 and
```
Comment (cmbuckley) [inline Terraform_modules/vpc_subnets/variables.tf:49]: `i` starts at 101 for public subnets

File: Terraform_modules/vpc_subnets/variables.tf [hunk 4]
```diff
@@ -1,54 +1,77 @@
...
 
 variable "num_private_subnets" {
   type        = number
-  description = "Number indicating how many public subnets you want"
+  description = <<EOT
+                This is a number specifying how many private subnets you want. Not specifying this will result
+                in `x` private subnets where `x` is the number of az zones. If the number specified is greater than
+                the number of Availability Zones (AZs) the public subnets will be spread out evenly over the
+                available AZs. The CIDR values used are of the form `10.0.{i}.0/24` where `i` starts at 1 and
+                increases by 1 for each public subnet.
```
[the remark this answers is not in the document]
Comment (JoeCSykes) [reply Terraform_modules/vpc_subnets/variables.tf:62]: not sure what you mean with this comment

File: Terraform_modules/vpc_subnets/variables.tf (hunk 3 above)
Comment (JoeCSykes) [reply Terraform_modules/vpc_subnets/variables.tf:49]: `i` starts from 101 for private subnets. I will make the changes for this in the variables and the README

File: Terraform_modules/vpc_subnets/variables.tf (hunk 4 above)
[the remark this answers is not in the document]
Comment (JoeCSykes) [reply Terraform_modules/vpc_subnets/variables.tf]: I believe I have sorted this

File: Terraform_modules/vpc_subnets/variables.tf (hunk 3 above)
Comment (JoeCSykes) [reply Terraform_modules/vpc_subnets/variables.tf:49]: I believe I have sorted this

**Comment by JoeCSykes** [comment]
> Not part of the diff here, but I would advise marking the username/password outputs of rds_serverless_cluster as `sensitive = true`.

I have changed this. Does this need to be shown in the README output table? If so, how do you want me to show it? i.e. in the desc or in a new column?

File: Terraform_modules/vpc_subnets/variables.tf (hunk 3 above)
Comment (cmbuckley) [reply Terraform_modules/vpc_subnets/variables.tf:49]: My bad, but looks correct now

**Comment by JoeCSykes** [comment]
@robg-answer I have set up terraform.docs.yml files for each module so that the README.md can be generated using 'terraform-docs -c .terraform.docs.yml <path-to-module> > README.md`. Do you thing it is worth running this command in your github action to autogenerate any changes in the README.md files?

**Comment by cmbuckley** [comment]
Could you update this to use a single `.terraform-docs.yml` at the root of the repo? You can fence the auto-generated part of the readme like this:

```md
<!-- BEGIN_TF_DOCS -->
...
<!-- END_TF_DOCS -->
```

and terraform-doc will inject updates into the file, if you also add config like this:

```yml
output:
  file: README.md
```

You might find you don't even need the config file when using the action as the defaults do the job:

```yml
    - name: Run terraform-docs and commit changes
      uses: terraform-docs/gh-actions@v1.0.0
      with:
        working-dir: "Terraform_modules/${{ matrix.module }}"
        output-method: inject
        git-push: "true"
```

File: .terraform.docs.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+formatter: "md"
+
+content: |-
```
Comment (cmbuckley) [inline .terraform.docs.yml:3]: I expect this content section can be removed as this is the default behaviour

File: .github/workflows/terraform.yml [hunk 6]
```diff
@@ -40,3 +40,11 @@ jobs:
         id: validate
         if: github.event_name == 'pull_request'
         run: terraform validate
+
+      - name: Update README Files
+        uses: terraform-docs/gh-actions@v1.0.0
+        with:
+          working-dir: "Terraform_modules/${{ matrix.module }}"
+          config-file: "../../.terraform.docs.yml"
```
Comment (cmbuckley) [inline .github/workflows/terraform.yml:48]: The conventional name for the file is `.terraform-docs.yml`

Comment (JoeCSykes) [reply .github/workflows/terraform.yml]: changed

File: .terraform.docs.yml (hunk 5 above)
Comment (JoeCSykes) [reply .terraform.docs.yml]: removed

**Comment by robg-answer** [comment]
@cmbuckley How does this know to recursively update all of the README.md's?

**Comment by cmbuckley** [comment]
@robg-answer it uses the `find-dir` input of the action [here](https://github.com/AnswerConsulting/AnswerKing-Infrastructure/pull/12/commits/dbe4156d6ec1f393d1c0b25c6d4d47d8eb3153ed#diff-10eac00f40a120814e0c56ca5bf5a61c8b335acbdeb977b72207606aa3daea4dR54).

Check the action docs: https://github.com/terraform-docs/gh-actions#inputs

**Comment by JoeCSykes** [comment]
how come this is picking up changes to the .idea folder when this is in the .gitignore file?

**Comment by cmbuckley** [comment]
Updated that now - `.idea/` will ignore recursively, but `.idea/*` is seen as a glob match only at the top-level

**Comment by JoeCSykes** [comment]
I am happy for this to be merged now

**Pull request merged**